lib/puppet_x/puppetlabs/azure/provider_arm.rb

```diff
@@ -107,9 +107,15 @@ def start_vm(machine)
 
         def get_all_vms # rubocop:disable Metrics/AbcSize
           begin
-            vms = ProviderArm.compute_client.virtual_machines.list_all.value!.body.value
+            vms = []
+            result = ProviderArm.compute_client.virtual_machines.list_all
+            vms += result.value
+            while ! result.next_link.nil? and ! result.next_link.empty? do
```

Comment:
See https://github.com/Azure/azure-sdk-ruby/blob/ARM-0.3.1/resource_management/azure_mgmt_compute/lib/azure_mgmt_compute/virtual_machines.rb#L812-L814 and my incorrect assumptions about https://tickets.puppetlabs.com/browse/FM-5149